pkg/cpio/fs_unix.go

```diff
@@ -144,11 +148,23 @@ func CreateFileInRoot(f Record, rootDir string, forcePriv bool) error {
 		return os.Symlink(string(content), f.Name)
 
 	case os.FileMode(0):
+		// Check the inode. If we have seen it before, create a hardlink
+		// and return. cpio files always have the contents with the
+		// first link.
+		Debug("New file ino is %d", f.Ino)
+		if of, ok := inodes[f.Ino]; ok {
+			if err := os.Link(of.Name, f.Name); err != nil {
+				return err
+			}
+		}
 		nf, err := os.Create(f.Name)
 		if err != nil {
 			return err
 		}
+		Debug("Created inode %v", f.Ino)
```

Comment:
Can you remove these debug fns?

Reply:
well ... they only run if you set the Debug to a non-empty function, and they are sure handy ... is it any harm to leave them? this hard link bit was a bit trickier than I expected.

Reply:
but if you want them gone, I'll yank them, strace is just as good now.